planex/extract.py

```diff
@@ -98,7 +98,7 @@ def archive_root(tar):
     """
     Return the name of the top level directory of the tarball
     """
-    if tar.firstmember.type == tarfile.DIRTYPE:
+    if len(tar.members) == 1 and tar.members[0].isdir():
```

Comment:
What we want to differentiate here is the following:
- tarball with one top level directory which contains all the sources
- tarball in which the sources are splattered at the top level without any directory.
We can only have these two cases.

An example of the first is:
tarball: test.tar
content: 
    test/a
    test/b
Instead, the second case looks like:
tarball: test.tar
content: a, b